text_adventure.py

```diff
@@ -0,0 +1,190 @@
+
+    @property
+    def contents(self):
+        if len(self.creatures) == 0:
+            return "There is nothing in this room"
+        return f"In this room is {[str(i) for i in self.creatures]}."
+
+class Creature:
+    def __init__(self, name, class_name, desc, health, damage, damage_range):
+        self.name = name
+        self.class_name = class_name
+        self.desc = desc
+        self.max_health = health
+        self.health = health
+        self.damage = damage
+        self.damage_range = damage_range
+    
+    def attack(self, target):
+        damage_randomized = self.damage * random.randrange(*self.damage_range)
+        target.health -= damage_randomized
+        print(f"The {self.name} hit you for {damage_randomized} damage!")
+
+    def __str__(self):
+        return self.class_name
+
+    def get_desc(self):
+        if self.health >= self.max_health:
+            return self.class_name + "\n" + self.desc
+        else:
+            health_line = f"{self.health}/{self.max_health} HP"
+            return self.class_name + "\n" + self.desc + "\n" + health_line
+    
+class Player:
+    def __init__(self, name, starting_room, health=25, damage=1, damage_range=(1,4), inventory=["Sword"]):
+        self.name = name
+        self.current_room = starting_room
+        self.max_health = health
+        self.health = health
+        self.damage = damage
+        self.damage_range = damage_range
+        self.inventory = inventory
+        print(f"You enter {self.current_room.name}. {self.current_room.contents}")
+
+    def attack(self, target):
+        damage_randomized = self.damage * random.randrange(*self.damage_range)
+        target.health -= damage_randomized
+        print(f"You hit the {target.name} for {damage_randomized} damage!")
+
+    def move_to_room(self, room_name):
+        for room in self.current_room.adjacent_rooms:
+            if room.name == room_name:
+                self.current_room = room
+                print(f"You have moved to {room_name}.")
+                print(self.current_room.contents)
+                return
+        print(f"{room_name} is not adjacent to your current room.")
+    
+    def heal(self):
+        heal_amount = random.randrange(3,5)
+        if self.health + heal_amount > self.max_health:
+            heal_amount = self.max_health - self.health
+        self.health += heal_amount
+        print(f"You have healed for {heal_amount} HP.")
+
+class Goblin(Creature):
+    def __init__(self, name, class_name="goblin", desc="A small green creature", health=random.randrange(5, 11), damage=1, damage_range=(1,3)):
+        super().__init__(name, class_name, desc, health, damage, damage_range)
+
+entrance = Room("entrance")
+goblin1 = Goblin("goblin")
+entrance.add_creature(goblin1)
+
+dungeon = Room("dungeon")
+goblin2 = Goblin("goblin")
+dungeon.add_creature(goblin2)
+
+entrance.adjacent_rooms.append(dungeon)
+dungeon.adjacent_rooms.append(entrance)
```

Comment:
Imagine making a whole map like this, how are you going to check that it makes logical sense?  (hint: graphs!)